File: frontend/src/services/clienteService.js
/**
 * GreenCore — Sistema de gestion de invernadero
 * Servicio HTTP para la entidad Cliente. Consume /api/v1/clientes.
 *
 * @module services/clienteService
 * @version 1.0.0
 */
import api from '../api/axiosInstance'

const resourceUrl = '/clientes'

export function getAllClientes() {
  return api.get(resourceUrl).then((r) => r.data)
}

export function getCliente(id) {
  return api.get(`${resourceUrl}/${id}`).then((r) => r.data)
}

export function createCliente(payload) {
  return api.post(resourceUrl, payload).then((r) => r.data)
}

export function updateCliente(id, payload) {
  return api.put(`${resourceUrl}/${id}`, payload).then((r) => r.data)
}

export function deleteCliente(id) {
  return api.delete(`${resourceUrl}/${id}`).then((r) => r.data)
}

File: backend/src/main/java/com/greencore/service/ClienteService.java
/**
 * GreenCore — Sistema de gestion de invernadero
 * Servicio de logica de negocio para la entidad Cliente.
 *
 * @author GreenCore Team
 * @version 1.0.0
 */
package com.greencore.service;

import com.greencore.model.Cliente;
import com.greencore.repository.ClienteRepository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.Optional;

@Service
@Transactional
public class ClienteService {

    private final ClienteRepository repository;

    public ClienteService(ClienteRepository repository) {
        this.repository = repository;
    }

    public List<Cliente> findAll() {
        return repository.findAll();
    }

    public Optional<Cliente> findById(Long id) {
        return repository.findById(id);
    }

    public Cliente save(Cliente entity) {
        return repository.save(entity);
    }

    public void delete(Long id) {
        repository.deleteById(id);
    }
}